### ISSUE TITLE
Command aliases

### ISSUE THREAD

**Issue by raine** [yargs/yargs#121]
Is there a way to add aliases to commands, similarly to `npm`?

That would be very useful.

**Comment by bcoe** [comment]
+1 to this feature, any thoughts about how they would be represented? would the aliases be added into `_`?

**Comment by raine** [comment]
You mean if `_` should have the alias or command that alias points to?

**Comment by bcoe** [comment]
@raine it feels a bit weird that the aliases would magically show up in `_`, I was thinking perhaps there'd be a new field on `argv`... perhaps? `agv._commandMap`? this could be populated like so:

```js
{
  command: ["alias-1", "alias-2"]
  command2: ["alias"],
  command3: []
}
```

What do you think?

**Comment by mnquintana** [comment]
This would be an awesome addition. I'm not sure what kind of workaround would work with the current API – it seems like you'd have to just add a separate command that happens to run the same function as another command? But that's not ideal, since yargs would display them as separate commands to the user, instead of as aliases for the same command.

**Comment by dogancelik** [comment]
:+1: This would be nice.

**Comment by vjpr** [comment]
+1

**Comment by donaldpipowitch** [comment]
@bcoe Was this fixed with the 4.x rebuild of commands?

**Comment by firstdoit** [comment]
I was surprised to understand that the `alias` function is only for options. :(
Such a common use case. I'll be forced to manually replace the alias before parsing with yargs.

**Comment by nexdrew** [comment]
@firstdoit If you use a module/object for your command, you can use an (undocumented) alias like:

```js
var cmd = require('./cmd') // or { ... }
yargs
  .command(cmd) // regular command
  .command('alias', false, cmd) // alias
  .argv
```

Or you can take an approach similar to @donaldpipowitch to make your alias(es) documented (included in help text), but commands in 4.x should define a builder and handler instead of just a handler.

Otherwise, command aliases is on my roadmap and I'll probably be able to hack something up in the next couple weeks.

**Comment by firstdoit** [comment]
@nexdrew nice, thanks, this works for now. 

Just for context, what I'm actually doing:

```
getCommands().forEach((c) => {
  const {command, alias, describe, builder} = c
  log.debug('Add command', command, alias, describe, builder)
  yargs.command(command, describe, builder)
  if (alias) {
    log.debug('Add alias', alias, 'for command', command)
    yargs.command(alias, false)
  }
})
```

**Comment by nexdrew** [comment]
@firstdoit Ok cool. I think if you expand your last statement there and pass `builder` for the alias, yargs should be able to handle `alias` just like `command`.

**Comment by firstdoit** [comment]
@nexdrew thanks for the help, but I'm going to have to try `commander`.

Apparently the support for sub commands is not very strong in `yargs`, and this is giving me all sorts of problems. If you want to take a look, here's where I'm at right now: https://github.com/vtex/toolbelt/blob/ed2aafd8ed7d01bfacffcfb58f7f24995e81982c/src/cli.js

Each module can define their own commands. That's why my case is trickier. But maybe you'll have some insights I'm not having.

**Comment by nexdrew** [comment]
@firstdoit Interesting! I could be wrong, but this looks like a great use-case for #494, which would allow you to define each subcommand as their own modules in a convenient directory structure. If you have time, take a look and let me know what you think. My recent focus on yargs is better support for nested command CLIs, so any feedback you have is welcomed!

**Comment by nexdrew** [comment]
@firstdoit In the meantime, yargs 4.x _should_ support subcommands with explicit `return yargs.command(...)` calls within the `builder` function of a command. Have you given that a shot yet?

**Comment by firstdoit** [comment]
Hey, @nexdrew :) no, I haven't .

I ended up building this small lib to quickly tend to my specific needs: https://github.com/vtex/findhelp

Very much WIP, but I'd love any feedback if you are feeling curious 😄

**Comment by nexdrew** [comment]
This is finally addressed in #647. I'll ping back here if/when it lands and is released.

**Comment by bcoe** [comment]
@raine @firstdoit @vjpr we've landed and released command aliases \o/

@nexdrew thanks for the amazing work on this feature.
